hal/armv7a/zynq7000/zynq.c

```diff
@@ -146,52 +151,52 @@ static int _zynq_setDevClk(int dev, char divisor0, char divisor1, char srcsel, c
 	switch (dev) {
 		case pctl_ctrl_usb0_clk:
 		case pctl_ctrl_usb1_clk:
-			id = dev - pctl_ctrl_usb0_clk;
-			*(zynq_common.slcr + slcr_usb0_clk_ctrl + id) = (*(zynq_common.slcr + pctl_ctrl_usb0_clk + id) & ~0x00000070) | (srcsel & 0x7) << 4;
+			id = dev - (u32)pctl_ctrl_usb0_clk;
+			*(zynq_common.slcr + slcr_usb0_clk_ctrl + id) = (*(zynq_common.slcr + pctl_ctrl_usb0_clk + id) & ~0x00000070U) | ((u32)srcsel & 0x7U) << 4;
 			break;
 
 		case pctl_ctrl_gem0_rclk:
 		case pctl_ctrl_gem1_rclk:
-			id = dev - pctl_ctrl_gem0_rclk;
-			*(zynq_common.slcr + slcr_gem0_rclk_ctrl + id) = (*(zynq_common.slcr + pctl_ctrl_gem0_rclk + id) & ~0x00000011) | (!!clkact0) |
-				((!!srcsel) << 4);
+			id = dev - (u32)pctl_ctrl_gem0_rclk;
+			*(zynq_common.slcr + slcr_gem0_rclk_ctrl + id) = (*(zynq_common.slcr + pctl_ctrl_gem0_rclk + id) & ~0x00000011U) | (clkact0 == 0U ? 0UL : 1UL) |
```

Comment:
here and in all below cases, as it's u32 and not used with bitshifts:

```suggestion
			*(zynq_common.slcr + slcr_gem0_rclk_ctrl + id) = (*(zynq_common.slcr + pctl_ctrl_gem0_rclk + id) & ~0x00000011U) | (clkact0 == 0U ? 0U : 1U) |
```